frontend/src/components/pages/search/hooks/useTrekResults.ts

```diff
@@ -77,8 +77,8 @@ export const useTrekResults = (
     hasNextPage,
     hasPreviousPage,
     isFetchingNextPage,
-  } = useInfiniteQuery<SearchResults, Error>(
-    [
+  } = useInfiniteQuery<SearchResults, Error>({
+    queryKey: [
       'trekResults',
       JSON.stringify(parsedFiltersState),
```

Comment:
```suggestion
      parsedFiltersState,
```

TanStack Query do it itself already: https://tanstack.com/query/latest/docs/framework/react/guides/query-keys